pkg/apis/messaging/v1/subscribable_channelable_validation.go

```diff
@@ -32,11 +34,20 @@ func isChannelEmpty(f duckv1.KReference) bool {
 func isValidChannel(ctx context.Context, f duckv1.KReference) *apis.FieldError {
 	errs := f.Validate(ctx)
 
-	// Namespace field is disallowed
-	if f.Namespace != "" {
-		fe := apis.ErrDisallowedFields("namespace")
-		fe.Details = "only name, apiVersion and kind are supported fields"
-		errs = errs.Also(fe)
+	if feature.FromContext(ctx).IsEnabled(feature.CrossNamespaceEventLinks) {
+		var resourceInfo cn.ResourceInfo
+		var flag *feature.Store
+		crossNamespaceError := cn.CheckNamespace(ctx, resourceInfo, flag)
+		if crossNamespaceError != nil {
+			errs = errs.Also(crossNamespaceError)
+		}
+	} else {
+		// Namespace field is disallowed
+		if f.Namespace != "" {
+			fe := apis.ErrDisallowedFields("namespace")
+			fe.Details = "only name, apiVersion and kind are supported fields"
+			errs = errs.Also(fe)
+		}
```

Comment:
I wrote some code for the checks. Could you take a look when you have the time? @Cali0707 @pierDipi

Reply:
This looks good, we can also mention the possibility of using the `namespace` by enabling the feature flag in the error details, something like:

```
only name, apiVersion and kind are supported fields when " + feature.CrossNamespaceEventLinks + is disabled
```

Reply:
I think for this comment, we're missing the change in `fe.Details` string?